src/v/datalake/cloud_uploader.h

```diff
@@ -32,6 +32,17 @@ class cloud_uploader {
       retry_chain_node& rtc_parent,
       lazy_abort_source& lazy_abort_source);
 
+    // Upload multiple data files.
+    // so that, in case of error, we know which files failed to upload.
+    // On success all files are uploaded and local files are removed.
+    // On failure, all files are removed.
+    ss::future<
+      result<chunked_vector<coordinator::data_file>, data_writer_error>>
```

Comment:
Same feedback here w.r.t tight coupling with the coordinator::data_file

Reply:
Also if you changed the interface as I'm suggesting, the caller could orchestrate the parallel uploads instead